### DISCUSSION TITLE
询问 关于密钥清单入库的原理

### DISCUSSION THREAD

**Discussion by liduote** [pjy612/SteamManifestCache#374] [Question / Answer]
大佬能否写一下清单入库的原理？

手动将decryptionKey加入config.vdf
并将对应的manifest加入到depotcache可行吗？

**Comment by pjy612** [comment]
你列的其实就是 **解锁后** Steam 下载游戏 的流程 中的两个步骤

Steam解锁后，误认为你拥有 解锁的游戏了。

Steam 需要会尝试去下载，那么下载什么呢？（这两个步骤实际和解锁工具无关，单纯就是Steam下载的流程

要下载的东西 需要通过 depotid 和 manifest 去关联。（这是 Steam 的下载机制，没有所谓的原理和为什么

那么 你配置了哪些 depotid 他就会去尝试获取哪些depotid 的密钥和对应的清单。（如果你不配置 只解锁一个appid，那么自然是空包 不会下载任何东西

那么默认你有密钥获取权限吗？（没有
那么默认你有清单获取权限吗？（也没有

所以 密钥提前放好，是准备工作，密钥不存在的话 Steam 直接不去获取清单，因为清单和清单对应的资源需要密钥去解析。

**（清单是bt种子，密钥是种子对应的压缩包的解压密码）**

Steam 会先去 config.vdf 里面找密钥，找不到就请求服务器获取。都没有就失败弹错误信息（内容加密中
所以 密钥要备好！

有密钥之后，Steam会尝试下载清单，先找缓存depotcache，没有再请求服务器获取清单code，通过清单code下载清单（默认拿不到code）如果拿不到清单，就失败弹错误提示（清单不可用
所以 清单要备好。

密钥清单提前备好了，自然就不用走服务器了，自然就能下载了。。。

那么 如何 对 **解锁工具** 做配置，让它知道要解锁什么呢？
去自己用对应的 **解锁工具**，看它们的**文档和说明**

**Reply by liduote** [reply]
多谢大佬，讲的很清楚。我操作成功了。非常感谢🙏

**Reply by pjy612** [reply]
wiki 里面 有一些 更具体的介绍
